x/margin/keeper/msg_server.go

```diff
@@ -211,6 +211,18 @@ func (k msgServer) OpenLong(ctx sdk.Context, msg *types.MsgOpen) (*types.MTP, er
 		return nil, err
 	}
 
+	safetyFactor := k.GetSafetyFactor(ctx)
+
+	lr, err := k.UpdateMTPHealth(ctx, *mtp, pool)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if lr.LTE(safetyFactor) {
+		return nil, types.ErrMTPUnhealthy
+	}
+
```

Comment:
Do we want to be checking mtp health here or pool health after opening? There is a pool open threshold we could maybe check against.

Reply:
@timlind we had a call yesterday with a bunch of us and Barlin at blockscience and the decision was to base it on the posterior mtp health